### PR TITLE
switch-to-configuration-ng: fix clippy issues and run clippy in drv

### PR DESCRIPTION
**Pull request by r-vdp**
## Things done

Relaunch #340803, but also run clippy in the check phase. 

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sw/switch-to-configuration-ng/package.nix [hunk 1]
```diff
@@ -22,6 +24,15 @@ rustPlatform.buildRustPackage {
 
   env.SYSTEMD_DBUS_INTERFACE_DIR = "${buildPackages.systemd}/share/dbus-1/interfaces";
 
+  nativeCheckInputs = [
+    clippy
+  ];
+
+  preCheck = ''
+    echo "Running clippy..."
+    ${lib.getExe cargo} clippy -- -Dwarnings
```
Comment (jmbaur) [inline pkgs/by-name/sw/switch-to-configuration-ng/package.nix:33]:
```suggestion
    cargo clippy -- -Dwarnings
```

Bringing in the cargo that will run on the host platform will cause cross-compilation of stc-ng to no longer work.

**Comment by r-vdp** [comment]
Hmm, the test is failing. I will investigate later.

File: pkgs/by-name/sw/switch-to-configuration-ng/package.nix (hunk 1 above)
Comment (r-vdp) [reply pkgs/by-name/sw/switch-to-configuration-ng/package.nix]: Fixed!

**Comment by r-vdp** [comment]
@jmbaur I fixed the test failure (there was a `!` too much in a condition after a manual clippy fix), and also changed the test to use `runTest` after noticing that it used about 40GB of memory to eval, it only uses about half of that now and evals a lot faster.
The main difference is that `runTest` makes sure that nixpkgs is instantiated only once, while before we would create a new nixpkgs instance for every specialisation (and this test has a lot of those!).

**Comment by r-vdp** [comment]
@ofborg test switchTestNg

**Review by jmbaur — APPROVED** [review]

**Pull request merged**